plugins/flashrom/fu-flashrom-device.c

```diff
@@ -309,6 +312,38 @@ fu_flashrom_device_constructed(GObject *obj)
 	fu_device_add_instance_id(FU_DEVICE(self), "main-system-firmware");
 }
 
+static void
+fu_flashrom_device_get_property(GObject *object, guint prop_id, GValue *value, GParamSpec *pspec)
+{
+	FuFlashromDevice *self = FU_FLASHROM_DEVICE(object);
+	switch (prop_id) {
+	case PROP_REGION:
+		g_value_set_string(value, self->region);
+		break;
+	default:
+		G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+		break;
+	}
+}
+
+static void
+fu_flashrom_device_set_property(GObject *object,
+				guint prop_id,
+				const GValue *value,
+				GParamSpec *pspec)
+{
+	FuFlashromDevice *self = FU_FLASHROM_DEVICE(object);
+	switch (prop_id) {
+	case PROP_REGION:
+		g_free(self->region);
+		self->region = g_value_dup_string(value);
+		break;
+	default:
+		G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+		break;
+	}
+}
+
 static void
 fu_flashrom_device_finalize(GObject *object)
```

Comment:
don't you need a `g_free(self->region);` here too?

Reply:
Yes, thanks. And we need `g_free()` in  `fu_superio_device_finalize()` too. The lack of `g_free()` there confused me.